### PR TITLE
Added documentation for options available for Hadoop user selection and related refactoring

### PR DESCRIPTION
**Pull request by PramodSSImmaneni**
@davidyan74 please see

Also related to https://github.com/apache/apex-core/pull/363, please merge the Apex one first.

### CODE REVIEW

**Comment by PramodSSImmaneni** [comment]
@davidyan74 please see

File: docs/dtgateway_security.md [hunk 1]
```diff
@@ -417,6 +421,41 @@ authentication. The next settings are JPAM related settings. The setting ```serv
...
+To specify this behavior, use `SPECIFIED_USER` as the value for `STRATEGY`. In this mode, a specific user name can be specified for the Hadoop user and it is used no matter who the authenticated user is. The Hadoop user name is specified by an additional property
+
+```xml
+<property>
+  <name>dt.gateway.hadoop.user.name</name>
+  <value>username</value>
+</property>
+```
+
+Impersonation is needed here as well and the requirements mentioned in the `Authenticated user` section above apply here as well.
```
Comment (davidyan74) [inline docs/dtgateway_security.md]: "Authenticated user" should be a link.

File: docs/dtgateway_security.md [hunk 2]
```diff
@@ -417,6 +421,41 @@ authentication. The next settings are JPAM related settings. The setting ```serv
...
+
+### Authenticated user
+
+To specify this behavior, use `AUTH_USER` as the value for `STRATEGY`. As explained earlier, this leads to applications being launched with the same user name as the authenticated user. This is the default behavior even when the property is not specified.
+
+In Kerberos secure mode, DT Gateway, would still connect to Hadoop by using it's Kerberos credentials a.k.a of the service user (default `dtadmin`) but impersonate the authenticated user to launch the application. This impersonation requires additional configuration on Hadoop side and it is explained in the Hadoop Configuration sub-section under the Impersonation section in the Apache Apex security [document](https://apex.apache.org/docs/apex/security/).
+
+### Gateway user
+
+To specify this behavior, use `GATEWAY_USER` as the value for `STRATEGY`. In this mode, the Hadoop user is same as the DT Gateway service user (default `dtadmin`) and no impersonation is necessary.
```
Comment (davidyan74) [inline docs/dtgateway_security.md]: same -> the same

File: docs/dtgateway_security.md [hunk 3]
```diff
@@ -417,6 +421,41 @@ authentication. The next settings are JPAM related settings. The setting ```serv
...
+</property>
+```
+
+The `STRATEGY` in the property above identifies the mapping behavior and the following options are available
+
+### Authenticated user
+
+To specify this behavior, use `AUTH_USER` as the value for `STRATEGY`. As explained earlier, this leads to applications being launched with the same user name as the authenticated user. This is the default behavior even when the property is not specified.
+
+In Kerberos secure mode, DT Gateway, would still connect to Hadoop by using it's Kerberos credentials a.k.a of the service user (default `dtadmin`) but impersonate the authenticated user to launch the application. This impersonation requires additional configuration on Hadoop side and it is explained in the Hadoop Configuration sub-section under the Impersonation section in the Apache Apex security [document](https://apex.apache.org/docs/apex/security/).
```
Comment (davidyan74) [inline docs/dtgateway_security.md]: remove comma after DT Gateway. 
it's -> its
a.k.a. : not sure why this term is appropriate here.

**Comment by PramodSSImmaneni** [comment]
@davidyan74 addressed comments

**Comment by PramodSSImmaneni** [comment]
@davidyan74 squashed

**Pull request merged**